File: server/php/src/Presentation/Mappers/HostedCheckoutMapper.php
<?php

namespace OnlinePayments\ExampleApp\Presentation\Mappers;

use OnlinePayments\ExampleApp\Application\Domain\Enums\Common\Currency;
use OnlinePayments\ExampleApp\Application\Domain\Enums\Common\Language;
use OnlinePayments\ExampleApp\Application\DTOs\HostedCheckout\RequestDto as CreateHostedCheckoutRequestDto;
use OnlinePayments\ExampleApp\Application\DTOs\HostedCheckout\ResponseDto as CreateHostedCheckoutResponseDto;
use OnlinePayments\ExampleApp\Presentation\Models\HostedCheckout\Request as CreateHostedCheckoutRequest;
use OnlinePayments\ExampleApp\Presentation\Models\HostedCheckout\Response as CreateHostedCheckoutResponse;
use OnlinePayments\ExampleApp\Application\DTOs\GetPaymentByHostedCheckoutId\ResponseDto as GetPaymentByHostedCheckoutIdResponseDto;
use OnlinePayments\ExampleApp\Presentation\Models\GetPaymentByHostedCheckoutId\Response as CreateGetPaymentByHostedCheckoutIdResponse;

class HostedCheckoutMapper
{
    private const SMALLEST_UNIT = 100;

    public static function map(CreateHostedCheckoutRequest $request): CreateHostedCheckoutRequestDto
    {
        return new CreateHostedCheckoutRequestDto(
            amount: (int)($request->amount * self::SMALLEST_UNIT),
            currency: Currency::from($request->currency),
            language: Language::from($request->language),
            billingAddress: $request->billingAddress ?? null,
            shippingAddress: $request->shippingAddress ?? null
        );
    }

    public static function mapResponse(CreateHostedCheckoutResponseDto $response): CreateHostedCheckoutResponse
    {
        return new CreateHostedCheckoutResponse(
            hostedCheckoutId: $response->hostedCheckoutId,
            redirectUrl: $response->redirectUrl,
            returnMAC: $response->returnMAC,
            amount: $response->amount / self::SMALLEST_UNIT,
            currency: $response->currency
        );
    }

    public static function mapGetPaymentResponse(GetPaymentByHostedCheckoutIdResponseDto $response): CreateGetPaymentByHostedCheckoutIdResponse
    {
        $mapped = new CreateGetPaymentByHostedCheckoutIdResponse();
        $mapped->status = $response->status;
        $mapped->paymentId = $response->paymentId;

        return $mapped;
    }
}

File: server/python/app/presentation/mappers/hosted_checkout_mapper.py
from app.application.domain.enums.common.language import Language
from app.application.dtos.hosted_checkout.request_dto import CreateHostedCheckoutRequestDto
from app.application.dtos.hosted_checkout.response_dto import CreateHostedCheckoutResponseDto
from app.application.dtos.get_payment_by_hosted_checkout_id.response_dto import GetPaymentByHostedCheckoutIdResponseDto
from app.presentation.models.hosted_checkout.request import HostedCheckoutRequest
from app.presentation.models.hosted_checkout.response import HostedCheckoutResponse

SMALLEST_UNIT = 100

class HostedCheckoutPresentationMapper:

    @staticmethod
    def to_dto(request: HostedCheckoutRequest) -> CreateHostedCheckoutRequestDto:
        return CreateHostedCheckoutRequestDto(
            amount=int(request.amount * SMALLEST_UNIT),
            currency=request.currency,
            language=Language(request.language),
            billing_address=request.billing_address,
            shipping_address=request.shipping_address,
            redirect_url=request.redirect_url
        )

    @staticmethod
    def from_dto(response_dto: CreateHostedCheckoutResponseDto) -> HostedCheckoutResponse:
        return HostedCheckoutResponse(
            hosted_checkout_id=response_dto.hosted_checkout_id,
            redirect_url=response_dto.redirect_url,
            return_mac=response_dto.return_mac,
            amount=response_dto.amount / SMALLEST_UNIT,
            currency=response_dto.currency
        )

    @staticmethod
    def from_get_payment_dto(response_dto: GetPaymentByHostedCheckoutIdResponseDto):
        return {
            "status": response_dto.status,
            "payment_id": response_dto.payment_id
        }